influxql/query_manager.go

```diff
@@ -162,11 +223,51 @@ func (qm *defaultQueryManager) waitForQuery(qid uint64, timeout time.Duration, c
 
 	select {
 	case <-closing:
+		qm.mu.Lock()
+		query, ok := qm.queries[qid]
+		qm.mu.Unlock()
```

Comment:
I'll add this as a private function. I was starting to think I was duplicating that code too many times myself.